### DISCUSSION TITLE
Is 'JsonPatchDocument' supported in OData?

### DISCUSSION THREAD

**Discussion by julealgon** [OData/AspNetCoreOData#635] [Q&A]
Title is basically the question.

https://docs.microsoft.com/en-us/aspnet/core/web-api/jsonpatch?view=aspnetcore-6.0

`JsonPatchDocument<T>` is a bit more powerful and expressive than a `Delta<T>` in that it allows for arbitrary link manipulation and hierarchical changes.

I don't see any explicit mention or evidence anywhere that OData works with it, but I've also not tried myself.

If it doesn't, will it ever do? What is the recommended alternative? Batch calls?

**Comment by vigouredelaruse** [comment]
consider this discussion of patch requests to odata endpoints, from the spec
http://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#sec_DeepInsert

and this
http://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#sec_UpdateaCollectionofEntities

notice the spec enforces the paylaod, and Delta.Apply() is designed to work with such payloads and nothing else, as per the spec

however there's nothing to stop a controller from responding to an odata url, say a patch, and accepting a jsonpatch payload

so you'd have something like
```
[EnableQuery]
[HttpPatch]
public IActionResult JsonPatchWithModelState(
    [FromBody] JsonPatchDocument<Customer> patchDoc)
{
    if (patchDoc != null)
    {
        var customer = CreateCustomer();

        patchDoc.ApplyTo(customer, ModelState);
......
```

in this scenario unless you wanted to for some reason you would work with JsonPatchDocument<T>, not Delta<T> to generate the sql expression

yes?
